Sources/Sentry/SentryPerformanceTracker.m

```diff
@@ -0,0 +1,194 @@
+#import "SentryPerformanceTracker.h"
+#import "SentryHub.h"
+#import "SentryLog.h"
+#import "SentrySDK+Private.h"
+#import "SentryScope.h"
+#import "SentrySpan.h"
+#import "SentrySpanId.h"
+#import "SentrySpanProtocol.h"
+#import "SentryTracer.h"
+
+static NSString *const SENTRY_PERFORMANCE_TRACKER_SPANS = @"SENTRY_PERFORMANCE_TRACKER_SPANS";
+static NSString *const SENTRY_PERFORMANCE_TRACKER_ACTIVE_STACK
+    = @"SENTRY_PERFORMANCE_TRACKER_ACTIVE_STACK";
+
+/*
+ * Auxiliary class to store tracking information.
+ */
+@interface SentrySpanTracker : NSObject
+
+@property (nonatomic) id<SentrySpan> span;
+@property (nonatomic) BOOL finished;
+@property (nonatomic) SentrySpanStatus finishedStatus;
+@property (nonatomic, strong) NSMutableDictionary<SentrySpanId *, SentrySpanTracker *> *children;
+
+- (instancetype)initWithSpan:(id<SentrySpan>)span;
+- (void)markFinishedWithStatus:(SentrySpanStatus)status;
+
+@end
+
+@implementation SentrySpanTracker
+
+- (instancetype)initWithSpan:(id<SentrySpan>)span
+{
+    if (self = [super init]) {
+        self.span = span;
+        self.finished = false;
+        self.children = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+- (void)markFinishedWithStatus:(SentrySpanStatus)status
+{
+    self.finishedStatus = status;
+    self.finished = YES;
+}
+@end
+
+@implementation SentryPerformanceTracker
+
++ (instancetype)shared
+{
+    static SentryPerformanceTracker *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
+    return instance;
+}
+
+/**
+ * A dictionary of created spans in this thread using id as Key.
+ */
+- (NSMutableDictionary<SentrySpanId *, SentrySpanTracker *> *)spansForThread
+{
+    NSMutableDictionary *result =
+        [NSThread.currentThread.threadDictionary objectForKey:SENTRY_PERFORMANCE_TRACKER_SPANS];
+    if (result == nil) {
+        result = [[NSMutableDictionary alloc] init];
+        [NSThread.currentThread.threadDictionary setObject:result
+                                                    forKey:SENTRY_PERFORMANCE_TRACKER_SPANS];
+    }
+    return result;
+}
+
+/**
+ * A stack of active spans in this thread.
+ */
+- (NSMutableArray<SentrySpanTracker *> *)activeStackForThread
```

Comment:
`m`: I'm not sure if we should limit the stack only to the current thread. I might want to see spans for HTTP requests from background threads combined with the spans from a UIViewController.

Reply:
Having http requests capture in the same transaction would not be a problem with this, but I will remove this limitation because it is not helping as much as I thought it would.